rita_common/src/rita_loop/write_to_disk.rs

```diff
@@ -0,0 +1,121 @@
+            thread::sleep(saving_to_disk_frequency - start.elapsed());
+        }
+
+        let router_mem_small;
+        //settings
+        match old_settings.clone() {
+            OldSettingsType::RitaClientSettings(old_settings_client) => {
+                let new_settings = get_rita_client();
+
+                if old_settings_client != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    info!("Config/settings is being saved to");
+                }
+                match new_settings.network.device {
+                    Some(val) => router_mem_small = is_router_memory_small(&val),
+                    None => router_mem_small = false,
+                }
+            }
+            OldSettingsType::RitaExitSettingsStruct(old_settings_exit) => {
+                let new_settings = get_rita_exit();
+
+                if old_settings_exit != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    info!("Config/settings is being saved to");
+                }
+                match new_settings.network.device {
+                    Some(val) => router_mem_small = is_router_memory_small(&val),
+                    None => router_mem_small = false,
+                }
+            }
+        }
+
+        // debt keeper
+        if !router_mem_small {
+            let runner = AsyncSystem::new();
+            runner.block_on(async move {
+                if let Err(e) = send_debt_update() {
+                    warn!("Debt keeper update failed! {:?}", e);
+                }
+                info!("Config/settings is being saved to");
+            });
+        }
+
+        //usage tracker
+        let babel_port = settings::get_rita_common().network.babel_port;
+        trace!("Common tick!");
+
+        let res = tm_get_neighbors();
+        trace!("Currently open tunnels: {:?}", res);
+        let neighbors = res;
+        let neigh = Instant::now();
+        info!(
+            "GetNeighbors completed in {}s {}ms",
+            start.elapsed().as_secs(),
+            start.elapsed().subsec_millis()
+        );
+
+        if let Ok(mut stream) = open_babel_stream(babel_port, FAST_LOOP_TIMEOUT) {
+            if let Ok(babel_routes) = parse_routes(&mut stream) {
+                if let Err(e) = watch(babel_routes, &neighbors) {
+                    error!("Error for Rita common traffic watcher {}", e);
+                }
+                info!(
+                    "TrafficWatcher completed in {}s {}ms",
+                    neigh.elapsed().as_secs(),
+                    neigh.elapsed().subsec_millis()
+                );
+            }
+        }
+    });
+}
+pub fn is_router_memory_small(router_model: &str) -> bool {
```

Comment:
this function seems to be duplicated from rita_bin, we should call this one from rita_bin (the other review comment about adding a doc comment and renaming it still applies)